src/map/packet_system.cpp

```diff
@@ -2378,29 +2379,133 @@ void SmallPacket0x042(map_session_data_t* const PSession, CCharEntity* const PCh
-    // uint32  msg_size_total = data.ref<uint32>(0x0C); // The total length of the requested server message
-    uint32 msg_offset = data.ref<uint32>(0x10); // The offset to start obtaining the server message
-    // uint32  msg_request_len = data.ref<uint32>(0x14); // The total requested size of send to the client
-
-    std::string login_message = luautils::GetServerMessage(msg_language);
-
-    PChar->pushPacket(new CServerMessagePacket(login_message, msg_language, msg_timestamp, msg_offset));
-    PChar->pushPacket(new CCharSyncPacket(PChar));
+    uint8  msg_chunk       = data.ref<uint8>(0x04);  // The current chunk of the message to send (1 = start, 2 = rest of message)
+    uint8  msg_type        = data.ref<uint8>(0x06);  // 1 = Server message, 2 = Fishing Rank
+    uint8  msg_language    = data.ref<uint8>(0x07);  // Language request id (2 = English, 4 = French)
+    uint32 msg_timestamp   = data.ref<uint32>(0x08); // The message timestamp being requested
+    uint32 msg_offset      = data.ref<uint32>(0x10); // The offset to start obtaining the server message
+    uint32 msg_request_len = data.ref<uint32>(0x14); // The total requested size of send to the client
 
-    // TODO: kill player til theyre dead and bsod
-    const char* fmtQuery = "SELECT version_mismatch FROM accounts_sessions WHERE charid = %u";
-    int32       ret      = _sql->Query(fmtQuery, PChar->id);
-    if (ret != SQL_ERROR && _sql->NextRow() == SQL_SUCCESS)
+    // uint8  msg_unknown1   = data.ref<uint8>(0x05);  // Unknown always 0
+    // uint32 msg_size_total = data.ref<uint32>(0x0C); // The total length of the requested server message
+
+    if (msg_type == 1) // Standard Server Message
     {
-        if ((bool)_sql->GetUIntData(0))
+        std::string login_message = luautils::GetServerMessage(msg_language);
+
+        PChar->pushPacket(new CServerMessagePacket(login_message, msg_language, msg_timestamp, msg_offset));
+        PChar->pushPacket(new CCharSyncPacket(PChar));
+
+        // TODO: kill player til theyre dead and bsod
+        const char* fmtQuery = "SELECT version_mismatch FROM accounts_sessions WHERE charid = %u";
+        int32       ret      = _sql->Query(fmtQuery, PChar->id);
+        if (ret != SQL_ERROR && _sql->NextRow() == SQL_SUCCESS)
         {
-            PChar->pushPacket(new CChatMessagePacket(PChar, CHAT_MESSAGE_TYPE::MESSAGE_SYSTEM_1, "Server does not support this client version."));
+            if ((bool)_sql->GetUIntData(0))
+            {
+                PChar->pushPacket(new CChatMessagePacket(PChar, CHAT_MESSAGE_TYPE::MESSAGE_SYSTEM_1, "Server does not support this client version."));
+            }
+        }
+    }
+    else if (msg_type == 2) // Fish Ranking Packet
+    {
+        // The Message Chunk acts as a "sub-type" for the request
+        // 1 = First packet of ranking table
+        // 2 = Subsequent packet of ranking table
+        // 10 = ???
+        // 11 = ??? Prepare to withdraw?
```

Comment:
Are these unknown, or the actual messages (I don't know fishing at all)

Reply:
These are unknown.  I saw them in the packets, but there was no discernable difference with those msg_chunk values.